File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
   public:
    int maxArea(vector<int>& heights) {
        int n = heights.size();
        int l = 0, r = n - 1;
        int maxx = INT_MIN;
        while (l < r) {
            int temp;
            if (heights[l] < heights[r]) {
                temp = (r - l) * heights[l];
                l++;
            } else if (heights[l] > heights[r]) {
                temp = (r - l) * heights[r];
                r--;
            } else {
                temp = (r - l) * heights[r];
                l++;
                r--;
            }
            if (temp > maxx) {
                maxx = temp;
            }
        }
        return maxx;
    }
};

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        r = len(arr)-2
        maxx = arr[-1]
        arr[-1] = -1
        while(r >=0):
            if(maxx<arr[r]):
                maxx,arr[r] = arr[r], maxx
            else:
                arr[r] = maxx
            r-=1
        return arr